### PR TITLE
Classifying the ADNI dataset for Alzheimer's by GFNet - s4743425

### PR DESCRIPTION
**Pull request by HairyHaz110**
Hi Teaching Team,

This is a pull request for the final version of the COMP3710 report sem 2 2024 for Harrison Taylor (47434253).

Please find attached the implementation of Project Task 5. Classification of Alzheimer’s disease (normal and AD) of the ADNI brain data. This algorithm is a GFNet implementation with all results shown in the README.md file. This request includes the code of the algorithm and test script in the files:

 - modules.py: containing the source code of the components of the model. 
 - dataset.py: containing the data loader for loading and preprocessing the ADNI data.
 - train.py: containing the source code for training, validating, testing and saving the model. 
 - predict.py: shows examples of the usage of the trained model.
 -utils.py: to calculate the mean and std of the training images.
- assets folder: containing images of the training/validation/testing results as well as explainer images for the README.md
  
  This pull request adds this folder under the topic-recognition folder.

Also attached is the README.md documentation of the algorithm, which includes a summary of the model and results. This same file as a PDF has also been submitted in Turnitin.

Thank you, 
Harry.

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.7063. 

**Model and functions:**

- It correctly uses PyTorch to construct the GFNet models and functions.
- With data augmentation.

**Code design:** Good.

**Code comment and docstring:**

- Good code comments 
- Good function docstrings

**Difficulty:** Hard.

**Additional Comments:**

- Reasonable commits.
- Good ReadMe design but it would be better if there is a conclusion in the report.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request includes a clear description about the file structure .
No previous feedback incorporated . 

File Organizing: Well-organized files.

Commit Log:
Commit messages are progressive for the Recognition Problem solved using 4 files . 
Commits are regularly made, showing incremental development rather than bulk updates.

Documentation:
Code comments and docstrings  are included.
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Can add about Conclusion and  future improvements in the ReadMe.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. Uses Data Aug.</td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>0</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Harrison Taylor

**Comment by shakes76** [comment]
Well done!
